sunpy/net/dataretriever/sources/gong.py

```diff
@@ -0,0 +1,289 @@
+"""
+This module implements GONG Client.
+"""
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import urllib2
+import numpy as np
+
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+
+__all__ = ['GONGClient', 'FARSIDEClient']
+
+class GONGClient(GenericClient):
+    """
+    Returns a list of URLS to GONG files corresponding to value of input timerange.
+    URL source: `ftp://gong2.nso.edu/QR/` and `http://gong2.nso.edu/HA/haf/`.
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+
+    Instrument: Arguments are any one from ['bigbear', 'cerrotelolo',
+                'learmonth', 'maunaloa', 'teide', 'udaipur','tucson']
+
+    Physobs: Two arguments or none 'intensity' and 'los_magnetic_field'
+
+    GONGClient expects at least one argument from Physobs or Instrument
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
```

Comment:
This example should be changed to a time and date that actually has GONG data available.